pykilosort/gui/minor_gui_elements.py

```diff
@@ -0,0 +1,292 @@
+                       self.y_coords_label, self.y_coords_value,
+                       self.k_coords_label, self.k_coords_value,
+                       self.channel_map_label, self.channel_map_value,
+                       self.bad_channels_label, self.bad_channels_value,
+                       self.error_label]
+
+        button_layout = QtWidgets.QHBoxLayout()
+        for button in buttons:
+            button_layout.addWidget(button)
+
+        self.okay_button.setDisabled(True)
+
+        for widget in widget_list:
+            layout.addWidget(widget)
+
+        layout.addLayout(button_layout)
+
+        self.setLayout(layout)
+
+    def set_values_as_unchecked(self):
+        self.values_checked = False
+        self.okay_button.setDisabled(True)
+
+    def set_values_as_checked(self):
+        self.values_checked = True
+        self.okay_button.setDisabled(False)
+
+    def check_inputs(self):
+        try:
+            map_name = self.map_name_value.text()
+            assert len(map_name.split()) == 1
+
+            x_coords = eval(self.x_coords_value.text())
+            y_coords = eval(self.y_coords_value.text())
+
+            x_coords = np.array(x_coords, dtype=np.float64)
+            y_coords = np.array(y_coords, dtype=np.float64)
+
+            assert len(x_coords) == len(y_coords)
+
+            k_coords = self.k_coords_value.text()
+            if k_coords == "":
+
+                k_coords = np.array([], dtype=np.float64)
+            else:
+                k_coords = np.array(eval(k_coords), dtype=np.float64)
+                assert x_coords.size == k_coords.size
+
+            channel_map = self.channel_map_value.text()
+            if channel_map == "":
+                channel_map = np.arange(x_coords.size)
+            else:
+                channel_map = np.array(eval(channel_map), dtype=np.int32)
+                assert x_coords.size == channel_map.size
+                assert channel_map.size == np.unique(channel_map).size
+                assert np.amax(channel_map) < channel_map.size
+
+            bad_channels = self.bad_channels_value.text()
+            if bad_channels == "":
+                bad_channels = np.array([], dtype=np.int32)
+            else:
+                bad_channels = np.array(eval(bad_channels), dtype=np.int32)
+                assert bad_channels.size < x_coords.size
+
+        except Exception as e:
+            self.error_label.setText(str(e))
+            self.error_label.show()
+
+        else:
+            self.map_name = map_name
+            self.x_coords = x_coords.tolist()
+            self.y_coords = y_coords.tolist()
+            self.k_coords = k_coords.tolist()
+            self.channel_map = channel_map.tolist()
+            self.bad_channels = bad_channels.tolist()
+
+            self.set_values_as_checked()
+            self.error_label.hide()
+
+            self.construct_probe()
+
+    def construct_probe(self):
+        probe = Bunch()
```

Comment:
Ideally we should use the `Probe` from pykilosort/params.py.

But not necessary.